# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/security/CSPEvalChecker.cpp
# Commit: ddf4012a7652
# Full Hash: ddf4012a7652e36d144fc43bc99335276deeafbe
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-05-22 03:40:58
# Regressor Bug: 965637
# File Overlap Count: 1
# Description:
#   Bug 965637: Move CSP from Principal into Client, part 1: backend changes. r=mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27654
# ==============================================================================

diff -r 6c2047bc2f3e -r ddf4012a7652 dom/security/CSPEvalChecker.cpp
--- a/dom/security/CSPEvalChecker.cpp	Tue May 21 22:51:54 2019 +0000
+++ b/dom/security/CSPEvalChecker.cpp	Tue May 21 23:14:27 2019 +0000
@@ -123,12 +123,7 @@
     return NS_OK;
   }
 
-  nsCOMPtr<nsIContentSecurityPolicy> csp;
-  nsresult rv = doc->NodePrincipal()->GetCsp(getter_AddRefs(csp));
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    *aAllowEval = false;
-    return rv;
-  }
+  nsresult rv = NS_OK;
 
   // Get the calling location.
   uint32_t lineNum = 0;
@@ -139,6 +134,7 @@
     fileNameString.AssignLiteral("unknown");
   }
 
+  nsCOMPtr<nsIContentSecurityPolicy> csp = doc->GetCsp();
   rv = CheckInternal(csp, nullptr /* no CSPEventListener for window */,
                      doc->NodePrincipal(), aExpression, fileNameString, lineNum,
                      columnNum, aAllowEval);